features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -850,6 +850,17 @@ internal open class RumViewScope(
             .submit()
     }
 
+    private fun updateGlobalAttributes(sdkCore: InternalSdkCore, event: RumRawEvent): Map<String, Any?> {
+        if (!stopped && event !is RumRawEvent.StopView) {
```

Comment:
because as you explained in the description of the ticket this will propagate the global attribute in case the user started a new view. A new view start will stopped the previous one automatically sending a `stopView` if I remember correctly. If the user added a global attribute to be propagated with the new view at `start` the old view will get this new attribute also if the `stop` was not called before and it is triggered by the `start` of the new view.

Reply:
No, `StopView` event is created only from `RumMonitor#stopView`. If new view starts, it won't create `StopView` event under the hood of `RumMonitor#startView` (instead, view will be stopped if it receives `StartView`, so maybe you want to address this case?).

Reply:
oh...damn...this is so complicated, yup in that case I will update the ticket, keep the stopView update and do not update it in case of `StartView` instead.